src/prediction_model.py

```diff
@@ -0,0 +1,288 @@
+                                                             return_train_score=True, 
+                                                             scoring=scoring_dict)).agg(['mean', 'std']).round(3).T
+
+    # Define Random Forest Regressor (RFR) pipeline
+    pipe_rfr = make_pipeline(
+        preprocessor,
+        RandomForestRegressor(n_jobs=-1)
+    )
+
+    # Define RFR hyperparameter options for optimization
+    param_dist_rfr = {
+        "randomforestregressor__max_depth": np.arange(20, 100, 2),
+        "randomforestregressor__bootstrap": [True, False],
+        "randomforestregressor__min_samples_leaf": [1, 2, 4],
+        "randomforestregressor__min_samples_split": [2, 5, 10]
+    }
+
+    # Perform hyperparameter optimization for RFR
+    rfr_search = RandomizedSearchCV(
+        pipe_rfr, param_dist_rfr, n_iter=10, n_jobs=-1, return_train_score=True
+    )
+    rfr_search.fit(X_train.iloc[0:2000], y_train.iloc[0:2000])
+
+    # Define new pipeline for RFR using optimized hyperparameters
+    pipe_rfr_opt = make_pipeline(
+        preprocessor,
+        RandomForestRegressor(max_depth=rfr_search.best_params_["randomforestregressor__max_depth"],
+                             bootstrap=rfr_search.best_params_["randomforestregressor__bootstrap"],
+                             min_samples_leaf=rfr_search.best_params_["randomforestregressor__min_samples_leaf"],
+                             min_samples_split=rfr_search.best_params_["randomforestregressor__min_samples_split"],
+                             n_jobs=-1)
+    )
+
+    # Perform cross-validation and store results in data frame
+    cross_val_results["random_forest"] = pd.DataFrame(cross_validate(pipe_rfr_opt, X_train.iloc[0:2000], y_train.iloc[0:2000], return_train_score=True, scoring=scoring_dict)).agg(['mean', 'std']).round(3).T
+
+    # Drop uneeded columns from scoring dataframes (originally kept for analysis purposes)
+    cross_val_results["dummy_regressor"] = cross_val_results["dummy_regressor"].drop(columns="std").rename(columns={"mean": "Dummy_Regressor"})
+    cross_val_results["random_forest"] = cross_val_results["random_forest"].drop(columns="std").rename(columns={"mean": "Random_Forest"})
+    cross_val_results["ridge"] = cross_val_results["ridge"].drop(columns="std").rename(columns={"mean": "Ridge"})
+
+    # Combine all results into one results data frame and save it
+    cross_val_results_df = cross_val_results["dummy_regressor"].join(cross_val_results["ridge"], how="inner").join(cross_val_results["random_forest"], how="inner")
+    cross_val_results_df.to_csv(results_dir + "model_comparison_table.csv")
+
+    # Fit the optimal RFR model
+    pipe_rfr_opt.fit(X_train, y_train)
+
+    # Create dataframe for actual scores from the testing data and the predicted scores from the model
+    results_dict = {
+        "Actual Scores": y_test,
+        "Predicted Scores": pipe_rfr_opt.predict(X_test)
+    }
```

Comment:
Using `results_dict` here, we can just call `mean_absolute_percentage_error` and call `.score` on the test set again and save that to a DataFrame.